capstone/src/main/java/com/google/sps/servlets/FileUnzipServlet.java

```diff
@@ -0,0 +1,161 @@
+package com.google.servlets;
+
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ByteArrayInputStream;
+
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.Arrays;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import com.google.api.gax.paging.Page;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.MultipartConfig;
+
+
+@WebServlet("/unzip")
+@MultipartConfig
+public class FileUnzipServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    
+    // String userId = request.getParamter("userId");
```

Comment:
Same as above - if this is a temp hack to make things work, add a TODO / github issue referencer here.